src/test/java/com/beust/jcommander/JCommanderTest.java

```diff
@@ -348,6 +359,9 @@ public void hiddenSplitter() {
       @Parameter(names = "--extensions", splitter = HiddenParameterSplitter.class)
       public List<String> extensions;
     }
+    if (HiddenParameterSplitter.class.getConstructors().length == 0) {
+      return; // Compiler has optimised away the private constructor
```

Comment:
`Class.getConstructors()` does not include private constructors, so the returned array will always have a length of zero. The java compiler *does not* optimize away much; mostly inlining primitive `final` fields and eliminating dead code. Other
optimizations are performed at runtime by the JVM.